cmd/nimbess-cni.go

```diff
@@ -164,7 +202,54 @@ func cmdAdd(args *skel.CmdArgs) error {
 		CNIVersion: conf.CNIVersion,
 	}
 
-	// TODO: handle IPAM type and data
+	// Run the IPAM plugin and get back the config to apply
+	ipamRequest, err := ipam.ExecAdd(cniRequest.IpamType, args.StdinData)
+	if err != nil {
+		return err
+	}
+
+	// Invoke ipam del if err to avoid ip leak
+	defer func() {
```

Comment:
should this defer happen before line 207? That way you issue a delete in case the ExecAdd fails on line 206, to ensure the ipam was cleaned.

Reply:
True